Various improvements

Diggsey opened the pull request:
- Add `rustls` feature and associated builder methods
- Add user authentication method
- Statically (but lazily) initialize regexes
- Use an error type rather than strings

A caveat is that I haven't tested the user authentication yet, but should be able to do that soon.

Diggsey committed:
- Add `rustls` feature and associated builder methods
- Add user authentication method
- Statically (but lazily) initialize regexes
- Use an error type rather than strings
- Return structured data from auth methods

madebywelch commented:
@WillSewell Can we get this merged and deployed?

WillSewell commented:
👋 thanks for contributing to this repository. I'm afraid due to life circumstances I just haven't had the bandwidth to review PRs, and stay up to date with Rust in general.

For that reason I'm afraid I'm going to have to archive this repository without merging outstanding PRs (at least for the time being).

Apologies I didn't get this notice in sooner.

bradfier committed:
User Info watchlist cannot be null

Pusher returns an error (but otherwise works) when you try and
authenticate the client with a null watchlist. An empty list serves the
same purpose, so remove the optionality from this structure.
